src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1175,10 +1176,13 @@ public synchronized void afterMerge(OnGoingMerge merge) {
     }
 
 
-    private void commitIndexWriter(IndexWriter writer, long translogId) throws IOException {
+    private void commitIndexWriter(IndexWriter writer, Translog.TranslogGeneration commit) throws IOException {
```

Comment:
Can we remove the `Translog.TranslogGeneration commit` arg entirely and just use `translog.getGeneration()`?  Seems like all callers do that...